app/assets/javascripts/discourse/lib/plugin-api.js.es6

```diff
@@ -308,6 +308,16 @@ class PluginApi {
   }
 
   /**
+   * Exposes the widget update ability to plugins. Updates the widget
+   * registry for the given widget name to include the properties on args
+   * See `reopenWidget` in `discourse/widgets/widget` from more ifo.
+  **/
+
+  reopenWidget(name, args) {
```

Comment:
You'll also need to bump the API version, otherwise people don't have a version to tie this new function to.